main-ui/src/main/java/com/example/starter/base/impl/AppShell.java

```diff
@@ -12,6 +12,6 @@
  */
 @PWA(name = "Project Base for Vaadin", shortName = "Project Base", iconPath = "static/icons/icon.png")
 @Theme("my-theme")
-@Push(transport = Transport.LONG_POLLING)
+@Push
```

Comment:
Default push means webscockets.
It's a separate pleasure to configure websockets support generally in arbitrary OSGi container.
Even though in Karaf WS should work without problems I'm not sure that Vaadin app will be able to work with Karaf WS.
At least this needs to be checked. Have you verified that it works ?

Long polling works in any OSGi container since there is no need any special support for that.

Reply:
Yup, checked, works fine. Checked with brower inspector that websocket is in use and the views appear automatically when going through the "demo script" described in the README file on the top level of the project.

Reply:
Alright.

Reply:
An option might be to add a comment that "with certain http whiteboard implementations, one might need to downgrade the push channel to long polling" 🤔

Reply:
It's not related to the HTTP whiteboard impl.
WS is not a part of it. I don't know how to deploy WS support for felix-jetty e.g.